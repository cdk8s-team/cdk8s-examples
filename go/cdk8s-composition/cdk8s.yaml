app: go run .
language: go
imports:
  - k8s@1.26.0