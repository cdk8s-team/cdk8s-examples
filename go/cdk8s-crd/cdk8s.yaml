app: go run .
language: go
imports:
  - k8s@1.26.0
  - https://raw.githubusercontent.com/cert-manager/cert-manager/master/deploy/crds/crd-certificates.yaml