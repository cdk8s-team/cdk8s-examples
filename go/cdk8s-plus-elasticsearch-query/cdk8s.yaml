app: go run .
language: go
imports:
  - k8s@1.26.0
  - https://download.elastic.co/downloads/eck/1.1.2/all-in-one.yaml