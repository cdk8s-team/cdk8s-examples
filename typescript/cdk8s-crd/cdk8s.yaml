app: ts-node index.ts
language: typescript
imports:
  - https://raw.githubusercontent.com/cert-manager/cert-manager/master/deploy/crds/crd-certificates.yaml