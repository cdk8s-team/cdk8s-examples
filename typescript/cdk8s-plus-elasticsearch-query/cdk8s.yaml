app: ts-node index.ts
language: typescript
imports:
  - k8s@1.26.0
  - https://download.elastic.co/downloads/eck/1.1.2/all-in-one.yaml