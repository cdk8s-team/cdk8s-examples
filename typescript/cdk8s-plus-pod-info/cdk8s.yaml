app: ts-node index.ts
language: typescript
