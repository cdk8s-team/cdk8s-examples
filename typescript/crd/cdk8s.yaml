app: ts-node index.ts
language: typescript
imports:
  - k8s@1.26.0
  - https://raw.githubusercontent.com/cert-manager/cert-manager/master/deploy/crds/crd-certificates.yaml