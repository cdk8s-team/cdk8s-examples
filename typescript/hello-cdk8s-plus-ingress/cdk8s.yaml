app: ts-node index.ts
language: typescript
imports:
  - k8s@1.26.0