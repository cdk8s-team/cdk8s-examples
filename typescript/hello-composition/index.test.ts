import { HelloComposition } from './index';
import { Testing } from 'cdk8s';

test('snapshot', () => {
  const app = Testing.app();
  const chart = new HelloComposition(app, 'hello-k8s');

  expect(Testing.synth(chart)).toMatchSnapshot();
});