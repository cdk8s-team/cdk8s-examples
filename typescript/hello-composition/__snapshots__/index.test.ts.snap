// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
[
  {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": {
      "name": "hello-k8s-hello-service-c8eb0d8d",
    },
    "spec": {
      "ports": [
        {
          "port": 80,
          "targetPort": 8080,
        },
      ],
      "selector": {
        "app": "hello-k8s-hello-c84324f6",
      },
      "type": "LoadBalancer",
    },
  },
  {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": {
      "name": "hello-k8s-hello-deployment-c8a02ea8",
    },
    "spec": {
      "replicas": 2,
      "selector": {
        "matchLabels": {
          "app": "hello-k8s-hello-c84324f6",
        },
      },
      "template": {
        "metadata": {
          "labels": {
            "app": "hello-k8s-hello-c84324f6",
          },
        },
        "spec": {
          "containers": [
            {
              "image": "paulbouwer/hello-kubernetes:1.7",
              "name": "web",
              "ports": [
                {
                  "containerPort": 8080,
                },
              ],
            },
          ],
        },
      },
    },
  },
  {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": {
      "name": "hello-k8s-ghost-service-c8ef4fe1",
    },
    "spec": {
      "ports": [
        {
          "port": 80,
          "targetPort": 2368,
        },
      ],
      "selector": {
        "app": "hello-k8s-ghost-c8e5b471",
      },
      "type": "LoadBalancer",
    },
  },
  {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": {
      "name": "hello-k8s-ghost-deployment-c85c8033",
    },
    "spec": {
      "replicas": 1,
      "selector": {
        "matchLabels": {
          "app": "hello-k8s-ghost-c8e5b471",
        },
      },
      "template": {
        "metadata": {
          "labels": {
            "app": "hello-k8s-ghost-c8e5b471",
          },
        },
        "spec": {
          "containers": [
            {
              "image": "ghost",
              "name": "web",
              "ports": [
                {
                  "containerPort": 2368,
                },
              ],
            },
          ],
        },
      },
    },
  },
]
`;
